### PR TITLE
Button: Improve `disabled`-related prop descriptions

### PR DESCRIPTION
**Pull request by mirka**
## What?

In `Button`, adds some clarity in the prop descriptions for `__experimentalIsFocusable` and `disabled`.

## Why?

Some of the wording was a bit confusing.

### CODE REVIEW

File: packages/components/src/button/types.ts [hunk 1]
```diff
@@ -115,23 +115,27 @@ type BaseButtonProps = {
...
 	 */
 	disabled?: boolean;
 };
 
 type AnchorProps = {
 	/**
 	 * Whether the button is disabled.
-	 * If `true`, this will force a `button` element to be rendered.
+	 *
+	 * If `true`, this will force a `button` element to be rendered, even when an `href` is given.
```
Comment (tyxla) [inline packages/components/src/button/types.ts:138]: Can this even be `true`, given that the type is `false`? Feels like this entire line is redundant here.

**Review by tyxla — APPROVED** [review]
👍 Thanks!

File: packages/components/src/button/types.ts (hunk 1 above)
Comment (mirka) [reply packages/components/src/button/types.ts:138]: Good question. In theory, yes it seems a bit wrong. In reality, the props types/descriptions for the `<a>` case and the `<button>` case are union-ed into a polymorphic type, and in most scenarios where this description can be shown to a developer I think this extra clarification is always relevant. For example, in the Storybook props table, or in IntelliSense (including parts of a codebase that are not type checked).

Basically even in an instance like `<Button href="foo" disabled={ false }>`, it's still useful to know that flipping the `disabled` to true would force the Button into a `<button>` element.
